### DISCUSSION TITLE
Web Origins Field Disabled When Selecting OAuth 2.0 Device Authorization Grant in Keycloak 25.0.4

### DISCUSSION THREAD

**Discussion by MAM-SYS** [keycloak/keycloak#32927] [Archive]
In Keycloak version 25.0.4, when selecting only the OAuth 2.0 Device Authorization Grant, the 'Web Origins' field in the admin UI gets disabled. Could you please help me understand why this happens?
![Screenshot 2024-09-14 at 2 53 01 PM](https://github.com/user-attachments/assets/0d72f069-fecb-4f99-a1bd-d8432df0c0b2)

I'm experiencing a CORS issue where the web origins haven't been properly configured in the Device Authorization Grant. This happens when attempting to call the following endpoint: `https://<domain>/realms/<realm>/protocol/openid-connect/auth/device`.

**Comment by darius-m** [comment]
I'd assume it's because the device authorization grant is meant for devices that don't have the required means for interactive authentication (e.g., a display or keyboard and mouse). Since it does not redirect the user from any application to Keycloak (you have to access the URL manually in a browser and enter a code, instead of being redirected automatically from a client application), there is no origin for the request that can be validated.

**Comment by honeynoorbakhsh** [comment]
I ran into the same problem...
